apollo-router/src/spec/query.rs

```diff
@@ -751,11 +725,15 @@ impl Query {
                         continue;
                     }
 
-                    if let Some(fragment) = self.fragments.get(name) {
+                    if let Some(Fragment {
+                        type_condition,
+                        selection_set,
+                    }) = self.fragments.get(name)
```

Comment:
This is a nonessential change, but I like that we use all fields from the Fragment type, so I made explicit restructuring here.